### PR TITLE
Handle no history case in RatioBasedEstimator

### PR DESCRIPTION
**Pull request by rubanm**
This was a regression from https://github.com/twitter/scalding/pull/1358/

Manually verified this handles the no history case correctly.

### CODE REVIEW

**Comment by Gabriel439** [comment]
:ship:

**Comment by sid-kap** [comment]
Looks good to me!

File: scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/RatioBasedEstimator.scala [hunk 1]
```diff
@@ -48,6 +48,9 @@ abstract class RatioBasedEstimator extends ReducerEstimator {
     val threshold = RatioBasedEstimator.getInputRatioThreshold(conf)
 
     historyService.fetchHistory(info, maxHistory) match {
+      case Success(Nil) =>
```
Comment (isnotinvain) [inline scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/RatioBasedEstimator.scala]: does fetchHistory return a List or a Seq? I'm wondering if Success(Nil) will still match say, Success(Vector()) if the underlying Seq type ever changes?

Comment (sid-kap) [inline scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/RatioBasedEstimator.scala]: It returns a Try[Seq[FlowStepHistory]].
Scala Seqs match by contents, so List() == Seq(), but yeah I guess it's possible for that to change.

**Comment by isnotinvain** [comment]
+1

**Pull request merged**